apps/studio/components/interfaces/Database/Publications/SinglePublication.tsx

```diff
@@ -0,0 +1,161 @@
+
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import Table from 'components/to-be-cleaned/Table'
+import InformationBox from 'components/ui/InformationBox'
+import { useDatabasePublicationsQuery } from 'data/database-publications/database-publications-query'
+import { useDatabasePublicationUpdateMutation } from 'data/database-publications/database-publications-update-mutation'
+import { useCheckPermissions, usePermissionsLoaded } from 'hooks/misc/useCheckPermissions'
+import SkeletonTableRow from 'components/ui/SkeletonTableRow'
+import { AlertCircle } from 'lucide-react'
+
+interface PublicationEvent {
+  event: string
+  key: string
+  example: string
+}
+
+interface SinglePublicationViewProps {
+  publicationName: string
+}
+
+const SinglePublicationView = ({ publicationName }: SinglePublicationViewProps) => {
+  const { project } = useProjectContext()
+
+  const { data: publication, isLoading } = useDatabasePublicationsQuery(
+    {
+      projectRef: project?.ref,
+      connectionString: project?.connectionString,
+    },
+    {
+      select: (data) => data.find((pub) => pub.name === publicationName),
+    }
+  )
+
+  const { mutate: updatePublications } = useDatabasePublicationUpdateMutation({
+    onSuccess: () => {
+      toast.success('Successfully updated event')
+      setToggleListenEventValue(null)
+    },
+  })
+
+  const canUpdatePublications = useCheckPermissions(
+    PermissionAction.TENANT_SQL_ADMIN_WRITE,
+    'publications'
+  )
+  const isPermissionsLoaded = usePermissionsLoaded()
+
+  const publicationEvents: PublicationEvent[] = [
+    { event: 'Insert', key: 'publish_insert', example: 'Adding a new todo "Buy groceries"' },
+    { event: 'Update', key: 'publish_update', example: 'Marking a todo as completed' },
+    { event: 'Delete', key: 'publish_delete', example: 'Deleting the "Buy groceries" todo' },
+    { event: 'Truncate', key: 'publish_truncate', example: 'Clearing all todos from the table' },
+  ]
+
+  const [toggleListenEventValue, setToggleListenEventValue] = useState<{
+    publication: any
+    event: PublicationEvent
+    currentStatus: any
+  } | null>(null)
+
+  const toggleListenEvent = async () => {
+    if (!toggleListenEventValue || !project) return
+
+    const { publication, event, currentStatus } = toggleListenEventValue
+    const payload = {
+      projectRef: project.ref,
+      connectionString: project.connectionString,
+      id: publication.id,
+    } as any
+    payload[`publish_${event.event.toLowerCase()}`] = !currentStatus
+    updatePublications(payload)
+  }
+
+  return (
+    <>
+      {isPermissionsLoaded && !canUpdatePublications && (
+        <div className="mb-4">
+          <InformationBox
+            icon={<AlertCircle className="text-foreground-light" strokeWidth={2} />}
+            title="You need additional permissions to update database publications"
+          />
+        </div>
```

Comment:
we're looking to adopt a pattern where we avoid hiding UI, but instead disable UI

i'd opt to show the toggles still, but disable them instead - add a tooltip for each of them "You need additional perms..."

Reply:
Toggles are show in a disabled state. This pattern is taken from the existing publications page.

<img width="1106" alt="image" src="https://github.com/user-attachments/assets/195aad28-a55c-441a-906c-81afc6e96b3d">